### ISSUE TITLE
Integrating StarRocks with Metabase

### ISSUE THREAD

**Issue by igorvpcleao** [StarRocks/starrocks#41951]
**TL;DR**: it seems Metabase MariaDB/ MySQL driver tries to call `time_format(TIME, varchar)` as part of the `sync-timezone` routine, but StarRocks doesn't provide such function.

### Steps to reproduce the behavior (Required)
1. Start a Metabase instance (either locally or SaaS).
2. Connect Metabase with StarRocks by following https://docs.starrocks.io/docs/integrations/BI_integrations/Metabase/.
3. Trigger a database sync.

### Expected behavior (Required)
1. See StarRocks tables on Metabase.

### Real behavior (Required)
1. Metabase succeeds on validating the connection.
2. I cannot see any tables as part of the database connection I've just created.
3. By checking Metabase logs, I can see a failure while calling `sync-timezone` routine (it seems to be the cause, but not sure).

### StarRocks version (Required)
3.2.3-a40e2f8
 
### Metabase version
Metabase Cloud (v1.48.7)/ v0.48.7

### Metabase logs
```
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.task.sync-databases Starting sync task for Database 35.
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 WARN metabase.driver.mysql 

********************************************************************************
WARNING: Metabase only officially supports MySQL 5.7/MariaDB 10.2 and above.
All Metabase features may not work properly when using an unsupported version.
********************************************************************************

[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: Sync metadata for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: step 'sync-dbms-version' for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util FINISHED: step 'sync-dbms-version' for mysql Database 35 'StarRocks' (151.5 ms)
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: step 'sync-timezone' for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:01-03:00 WARN metabase.sync.util Error running step 'sync-timezone' for mysql Database 35 'StarRocks'
java.sql.SQLTransientConnectionException: (conn=601) Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
	at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
	at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
	at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1113)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.mysql$fn__106633.invokeStatic(mysql.clj:205)
	at metabase.driver.mysql$fn__106633.invoke(mysql.clj:195)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.sql_jdbc$fn__111186.invokeStatic(sql_jdbc.clj:76)
	at metabase.driver.sql_jdbc$fn__111186.invoke(sql_jdbc.clj:71)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invokeStatic(sync_timezone.clj:41)
	at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invoke(sync_timezone.clj:36)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.util$run_step_with_metadata$fn__57032.doInvoke(util.clj:425)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:150)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:146)
	at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:420)
	at metabase.sync.util$run_step_with_metadata.invoke(util.clj:415)
	at metabase.sync.util$run_sync_operation$fn__57111.invoke(util.clj:531)
	at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:529)
	at metabase.sync.util$run_sync_operation.invoke(util.clj:523)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__81408.invoke(sync_metadata.clj:63)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:191)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:184)
	at clojure.core$partial$fn__5910.invoke(core.clj:2647)
	at metabase.driver$fn__48779.invokeStatic(driver.clj:725)
	at metabase.driver$fn__48779.invoke(driver.clj:725)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__56973.invoke(util.clj:167)
	at metabase.sync.util$with_db_logging_disabled$fn__56970.invoke(util.clj:159)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
	at metabase.sync.util$with_start_and_finish_logging$fn__56957.invoke(util.clj:144)
	at metabase.sync.util$with_sync_events$fn__56952.invoke(util.clj:118)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__56939.invoke(util.clj:90)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:216)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:210)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:62)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:58)
	at metabase.task.sync_databases$sync_and_analyze_database_STAR__BANG_.invokeStatic(sync_databases.clj:94)
	at metabase.task.sync_databases$sync_and_analyze_database_STAR__BANG_.invoke(sync_databases.clj:77)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invokeStatic(sync_databases.clj:113)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invoke(sync_databases.clj:101)
	at metabase.task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:118)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
	... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
	... 60 more
```

**Comment by EEMaddy** [comment]
+1

**Comment by moweonlee** [comment]
Same issue on 
Starrocks : 3.2.2 
Metabase : 0.48.6

```
java.sql.SQLTransientConnectionException: (conn=3) Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
        at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
        at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
        at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
        at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
        at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
        at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1113)
        at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
        at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
        at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
        at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
        at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
        at metabase.driver.mysql$fn__102784.invokeStatic(mysql.clj:205)
        at metabase.driver.mysql$fn__102784.invoke(mysql.clj:195)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.driver.sql_jdbc$fn__105606.invokeStatic(sql_jdbc.clj:76)
        at metabase.driver.sql_jdbc$fn__105606.invoke(sql_jdbc.clj:71)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invokeStatic(sync_timezone.clj:41)
        at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invoke(sync_timezone.clj:36)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:669)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase.sync.util$run_step_with_metadata$fn__57009.doInvoke(util.clj:425)
        at clojure.lang.RestFn.invoke(RestFn.java:397)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
        at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:150)
        at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:146)
        at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:420)
        at metabase.sync.util$run_step_with_metadata.invoke(util.clj:415)
        at metabase.sync.util$run_sync_operation$fn__57088.invoke(util.clj:531)
        at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:529)
        at metabase.sync.util$run_sync_operation.invoke(util.clj:523)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__80792.invoke(sync_metadata.clj:63)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:191)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:184)
        at clojure.core$partial$fn__5910.invoke(core.clj:2647)
        at metabase.driver$fn__48768.invokeStatic(driver.clj:725)
        at metabase.driver$fn__48768.invoke(driver.clj:725)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at metabase.sync.util$sync_in_context$fn__56950.invoke(util.clj:167)
        at metabase.sync.util$with_db_logging_disabled$fn__56947.invoke(util.clj:159)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
        at metabase.sync.util$with_start_and_finish_logging$fn__56934.invoke(util.clj:144)
        at metabase.sync.util$with_sync_events$fn__56929.invoke(util.clj:118)
        at metabase.sync.util$with_duplicate_ops_prevented$fn__56916.invoke(util.clj:90)
        at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:216)
        at metabase.sync.util$do_sync_operation.invoke(util.clj:210)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:62)
        at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:58)
        at metabase.api.database$fn__97946$fn__97951.invoke(database.clj:998)
        at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
        ... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
        ... 60 more

```

**Comment by moweonlee** [comment]
@igorvpcleao 

I started to think that unsupported "time_format(TIME, varchar)" function may not the cause of this problem.

The reason that I think so is that I can see all tables in Starrocks 3.2.2 even it has same error output in the log below.

```
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
        at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
        ... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
        ... 60 more
```


The only change I found is version.
3.2.2 -> working.
3.2.3 -> not working.

In 3.2.3 , it doesn't look table scanning find anything in fingerprint-fields, classify-fields and classify-tables stage.

![image](https://github.com/StarRocks/starrocks/assets/16459517/6159bf93-f0ea-45c6-bb4c-2337598164ca)

But in 3.2.2, it successfully find tables and columns.

![image](https://github.com/StarRocks/starrocks/assets/16459517/4879fb57-ca45-4146-b17f-b39132fcde8f)

**Comment by moweonlee** [comment]
Table scanning process is working like below.

1) sync-dbms-version
2) sync-timezone : both 3.2.2, 3.2.3 have same issue on getting time zone information.
3) 'sync-tables'
4) 'sync-fields' 
5) 'sync-fks' 
6) 'sync-metabase-metadata' 
7) 'sync-table-privileges' 

8) 'Analyze data for mysql database 5 'Starrocks-N3R'
8-1) 'fingerprint-fields' 
8-2) 'classify-fields' 
8-3) 'classify-tables' 

In stage 8 , Metabase fail to read table metadata in version 3.2.3

**Comment by dorsegal** [comment]
Any chance to get it back open?

**Comment by dorsegal** [comment]
@stdpain we still have issues with it

**Comment by stdpain** [comment]
@dorsegal can u find the reason? or a unsupported SQL?

**Comment by dorsegal** [comment]
I don't know yet. I asked at metabase https://github.com/metabase/metabase/issues/50930#issuecomment-2572660019

**Comment by stdpain** [comment]
@dorsegal can u check the logs in fe/log/fe.warn.log and fe/log/fe.audit.log ?

**Comment by stdpain** [comment]
maybe metabase need the builtin table STATISTICS;
```
mysql> select count(*) from INFORMATION_SCHEMA.STATISTICS ;
+----------+
| count(*) |
+----------+
|        0 |
+----------+
1 row in set (0.00 sec)
```

**Comment by stdpain** [comment]
@dorsegal 
 can u help update the doc in https://docs.starrocks.io/docs/integrations/BI_integrations/Metabase ?

**Comment by stdpain** [comment]
docs update in https://github.com/StarRocks/starrocks/pull/54754.
